packages/agent/src/channel.ts

```diff
@@ -27,22 +27,7 @@ export abstract class BaseChannel implements Channel {
   abstract start(): void;
   abstract stop(): Promise<void>;
   abstract sendToRemote(message: AgentTransmitResponse): void;
-
-  async reloadConfig(definition: AgentChannel, endpoint: Endpoint): Promise<void> {
-    const previousEndpoint = this.endpoint;
-    this.definition = definition;
-    this.endpoint = endpoint;
-
-    this.log.info('Reloading config... Evaluating if channel needs to change address...');
-
-    if (needToRebindToPort(previousEndpoint, endpoint)) {
-      await this.stop();
-      this.start();
-      this.log.info(`Address changed: ${previousEndpoint.address} => ${endpoint.address}`);
-    } else {
-      this.log.info(`No address change needed. Listening at ${endpoint.address}`);
-    }
-  }
+  abstract reloadConfig(definition: AgentChannel, endpoint: Endpoint): Promise<void>;
```

Comment:
Made this method abstract; there are slight divergences / implementation-specific details between HL7 and DICOM channels already. 

The superclass method seemed to be a pre-mature factoring out of this method. Pushing the implementation back down to the subclass.